src/app/services/baw-api/baw-api.service.ts

```diff
@@ -255,11 +255,88 @@ export interface Paging {
   next?: string;
 }
 
+interface Combinations {
+  and?: InnerFilter;
+  or?: InnerFilter;
+  not?: InnerFilter;
+}
+
+interface Comparisons {
+  eq?: string | number;
+  equal?: string | number;
+  notEq?: string | number;
+  notEqual?: string | number;
+  lt?: number;
+  lessThan?: number;
+  notLt?: number;
+  notLessThan?: number;
+  gt?: number;
+  greaterThan?: number;
+  notGt?: number;
+  notGreaterThan?: number;
+  lteq?: number;
+  lessThanOrEqual?: number;
+  notLteq?: number;
+  notLessThanOrEqual?: number;
+  gteq?: number;
+  greaterThanOrEqual?: number;
+  notGteq?: number;
+  notGreaterThanOrEqual?: number;
+}
+
+type Range =
+  | string[]
+  | number[]
+  | {
+      interval?: string;
```

Comment:
interval has a narrow syntax that is acceptable: `(\[|\()[-+]?[0-9]*\.?[0-9]+,[-+]?[0-9]*\.?[0-9]+(\]|\))`

May be able to strongly type it?

Reply:
Regex typing is not currently possible: https://github.com/Microsoft/TypeScript/issues/6579.
I saw that its possible to strong type the value using the following brute force "hack":
```typescript
type Interval = "[0, 1]" | "[0, 2]" | ... | ["999", "999"];
```

Reply:
Yup we can't regex match the type. Could we just make it a type though, i.e a class?

Reply:
Do you want the class to type check the input at run time, or basically just be a string wrapper?

Reply:
It can do a bit of checking yeah